streaming/src/main/scala/org/apache/spark/streaming/util/RecurringTimer.scala

```diff
@@ -99,6 +99,8 @@ class RecurringTimer(clock: Clock, period: Long, callback: (Long) => Unit, name:
         nextTime += period
         logDebug("Callback for " + name + " called at time " + prevTime)
       }
+      clock.waitTillTime(nextTime)
```

Comment:
Do we really need to wait till the next time? You could just take a parameter specifying whether the method should be called on stop or not, and then pass the current time in.